File: educore-main/edu-main/scratch/fetch_openapi.ts
import fetch from 'node-fetch';
import * as fs from 'fs';
import * as path from 'path';

// Manual .env parser
function loadEnv() {
    const envPath = path.join(process.cwd(), '.env');
    if (fs.existsSync(envPath)) {
        const content = fs.readFileSync(envPath, 'utf8');
        const lines = content.split('\n');
        for (const line of lines) {
            const trimmedLine = line.trim();
            if (!trimmedLine || trimmedLine.startsWith('#')) continue;
            const eqIdx = trimmedLine.indexOf('=');
            if (eqIdx > 0) {
                const key = trimmedLine.substring(0, eqIdx).trim();
                let value = trimmedLine.substring(eqIdx + 1).trim();
                if ((value.startsWith('"') && value.endsWith('"')) || (value.startsWith("'") && value.endsWith("'"))) {
                    value = value.slice(1, -1);
                }
                process.env[key] = value;
            }
        }
    }
}

loadEnv();

const supabaseUrl = process.env.VITE_SUPABASE_URL;
const anonKey = process.env.VITE_SUPABASE_ANON_KEY;

async function run() {
    if (!supabaseUrl || !anonKey) {
        console.error('Missing env vars');
        return;
    }

    const response = await fetch(`${supabaseUrl}/rest/v1/?apikey=${anonKey}`);
    const data = await response.json();
    
    console.log('Exposed RPCs:');
    const paths = Object.keys((data as any).paths || {});
    const rpcs = paths.filter(p => p.startsWith('/rpc/'));
    rpcs.forEach(r => console.log(r));
}

run();

File: educore-main/edu-main/scratch/fetch_openapi.py
import urllib.request
import json
import os

def load_env():
    env = {}
    if os.path.exists('.env'):
        with open('.env', 'r', encoding='utf-8') as f:
            for line in f:
                line = line.strip()
                if not line or line.startswith('#'):
                    continue
                if '=' in line:
                    key, val = line.split('=', 1)
                    env[key.strip()] = val.strip().strip('"').strip("'")
    return env

def main():
    env = load_env()
    supabase_url = env.get('VITE_SUPABASE_URL')
    anon_key = env.get('VITE_SUPABASE_ANON_KEY')
    service_key = env.get('SUPABASE_SERVICE_ROLE_KEY')
    if not supabase_url:
        print("Missing environment variables.")
        return

    # Try service key first, fallback to anon key
    key = service_key or anon_key
    if not key:
        print("No API key found.")
        return

    url = f"{supabase_url}/rest/v1/"
    headers = {
        'apikey': key,
        'Authorization': f'Bearer {key}'
    }
    try:
        req = urllib.request.Request(url, headers=headers)
        with urllib.request.urlopen(req) as response:
            data = json.loads(response.read().decode('utf-8'))
            paths = data.get('paths', {})
            rpcs = [path for path in paths if path.startswith('/rpc/')]
            print("Exposed RPCs:")
            for rpc in sorted(rpcs):
                print(rpc)
    except Exception as e:
        print("Error fetching OpenAPI schema:", e)

if __name__ == '__main__':
    main()
